publish-npm/action.yml

```diff
@@ -0,0 +1,80 @@
+name: publish NPM
+
+inputs:
+  mvn_settings_filepath:
+    description: Filepath to the settings.xml file
+    default: '.github/maven.settings.xml'
+    required: false
+  nexus_username:
+    description: 'Nexus Username'
+    required: true
+  nexus_password:
+    description: 'Nexus password'
+    required: true
+  module_path:
+    description: 'Path to module'
+    default: './'
+  node_version:
+    description: 'Version of node to install on the host'
+    required: false
+    default: 'lts/*'
+
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node_version }}
+
+    - name: Set environment variables from parameters
+      shell: bash
+      run: |
+        echo "NEXUS_USERNAME=${{ inputs.nexus_username }}" >> $GITHUB_ENV
+        echo "NEXUS_PASSWORD=${{ inputs.nexus_password }}" >> $GITHUB_ENV
+        echo "MODULE_NAME=$(jq '.name' package.json | cut -d\" -f2)" >> $GITHUB_ENV
+        MODULE_VERSION=$(jq '.version' package.json | cut -d\" -f2)
+        echo "REPOSITORY_ID=$(jq '.jahia.maven.distributionManagement.snapshotRepository.id' package.json | cut -d\" -f2)" >> $GITHUB_ENV
+        echo "REPOSITORY_URL=$(jq '.jahia.maven.distributionManagement.snapshotRepository.url' package.json | cut -d\" -f2)" >> $GITHUB_ENV
+        echo "GROUP_ID=$(jq '.jahia.maven.groupId' package.json | cut -d\" -f2)" >> $GITHUB_ENV
+
+        if [[ $MODULE_VERSION == *-SNAPSHOT ]]; then
+          echo "MODULE_VERSION=$MODULE_VERSION" >> $GITHUB_ENV
+        else
+          echo "MODULE_OLD_VERSION=$MODULE_VERSION" >> $GITHUB_ENV
+          MODULE_VERSION+="-SNAPSHOT"
+          echo $MODULE_VERSION
+          echo "MODULE_VERSION=$MODULE_VERSION" >> $GITHUB_ENV
+        fi
+
+    # This step save the maven cache between runs
+    # More details can be found here: https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows
+    - name: Cache local Maven repository
+      uses: actions/cache@v3
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Prepare the module
+      shell: bash
+      run: yarn
```

Comment:
I believe you should be using `module_path` here, yarn is only going to work if the module itself is at the repository root.

Same for the other commands